Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -307,28 +310,15 @@ extension AWSClient {
         case .query:
             let data = try AWSShapeEncoder().encodeToJSONUTF8Data(input)
             var dict = try JSONSerializer().serializeToFlatDictionary(data)
+
             dict["Action"] = operationName
             dict["Version"] = apiVersion
 
-            var queryItems = [String]()
-            let keys = Array(dict.keys).sorted()
-
-            for key in keys {
-                if let value = dict[key] {
-                    queryItems.append("\(key)=\(value)")
-                }
+            if let urlEncodedQueryParams = urlEncodeQueryParams(fromDictionary: dict) {
+                body = .text(urlEncodedQueryParams)
             }
 
-            if let params = queryItems.joined(separator: "&").addingPercentEncoding(withAllowedCharacters: CharacterSet(charactersIn: String.uriAWSQueryAllowed.joined())) {
-
-                if path.contains("?") {
-                    path += "&" + params
-                } else {
-                    path += "?" + params
-                }
-
-                body = .text(params)
-            }
+            queryParams = queryParams.merging(dict) { $1 }
```

Comment:
is there ever a case where the Input.queryParams share a key with what is in the input and this merge will overwrite a key?